components/Paginator.tsx

```diff
@@ -54,14 +61,17 @@ const Paginator: React.FC<Props> = ({ page, setPage, numPosts }) => {
         <button
           className={cx(
             'transition duration-100 border-l dark:border-slate-700 font-bold rounded-r-lg hidden md:flex px-4 py-4 w-16 lg:w-32 items-center justify-center',
-            page === Math.ceil(numPosts / 8)
+            page === numPages
               ? 'bg-slate-50 dark:bg-gray-900/30 text-slate-400 dark:text-slate-500 cursor-not-allowed'
               : 'bg-white dark:bg-slate-800 hover:bg-sky-50 dark:hover:bg-slate-700 text-sky-500 dark:text-sky-400',
           )}
           onClick={() => {
-            if (page < Math.ceil(numPosts / 8)) {
-              setPage(page + 1);
-              scrollTo({ top: 0, left: 0, behavior: 'smooth' });
+            if (page < numPages) {
+              router.push(
+                language === 'en'
+                  ? `/posts/page/${page + 1}`
+                  : `/publicaciones/pagina/${page + 1}`,
```

Comment:
...looking at this again, there might be more styling difference than i first imagined, making a single component harder.  maybe it would need a `type: 'numbered' | 'prevNext'` prop as well.   ...or there could be two components and a helper function to calculate the url to a `string | null` or something like that. however you pull it off, i think:

1. they've got to be real links when they are not disabled
2. the tricky "is it disabled?" logic and duplicated path stuff should by DRY'd up

Reply:
...and another thought.  you could possibly just not render the disabled buttons, which could make it easier. i would want to make sure the flexbox centering of the numbered links didn't get dorked by the asymmetry, but it's an option to consider.